vertical-pod-autoscaler/pkg/recommender/model/aggregations_config.go

```diff
@@ -17,22 +17,23 @@ limitations under the License.
 package model
 
 import (
+	"flag"
 	"time"
 
 	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/util"
 )
 
 var (
 	// MemoryAggregationWindowLength is the length of the memory usage history
-	// aggregated by VPA, which is 8 days.
-	MemoryAggregationWindowLength = time.Hour * 8 * 24
+	// aggregated by VPA.
+	MemoryAggregationWindowLength = DefaultMemoryAggregationWindowLength
```

Comment:
The default assigned here is overriden anyway by the flag default.

Can we have one of two schemes (I'm using MemoryAggregationWindowLength as an example).

1. Directly here:
````
MemoryAggregationWindowLength = flag.Duration("memory-aggregation-window-length", DefaultMemoryAggregationWindowLength, 'The length of the memory usage history aggregated by VPA.')
````
change uses to *MemoryAggregationWindowLength


2. Separate flags defined here or in main.go
```
var(
RecommenderMemoryAggregationWindowLength = flag.Duration(&, "recommender-memory-aggregation-window-length", DefaultMemoryAggregationWindowLength, 'The length of the memory usage history aggregated by VPA.')
)
````
and here 
````
MemoryAggegationWindowLength = *RecommenderMemoryAggregationWindowLength
````
Although ideally we would have aggregationConfig struct with those values passed around.

Reply:
Sure. I could do that. I just went with the lexical change. I like the consistency of the first suggestion.

```
MemoryAggregationWindowLength = flag.Duration("memory-aggregation-window-length", DefaultMemoryAggregationWindowLength, 'The length of the memory usage history aggregated by VPA.')
```

I will make the changes.

Reply:
I bit the bullet and introduced the [`AggregationsConfig` struct](https://github.com/kubernetes/autoscaler/pull/2905/files#diff-7f21ec2575f476dc048bce486d6eec8dR25-R54), though still as a [singleton instance](https://github.com/kubernetes/autoscaler/pull/2905/files#diff-7f21ec2575f476dc048bce486d6eec8dR118) which is now initialised in the [main.go based on the command-line flags](https://github.com/kubernetes/autoscaler/pull/2905/files#diff-483c01ba6f34fd6f8c9d4944684cff98R67).

For the test-cases, this instance may not be initialised in which case it would be [initialised to the current defaults](https://github.com/kubernetes/autoscaler/pull/2905/files#diff-7f21ec2575f476dc048bce486d6eec8dR123).